ui/build/build.types.js

```diff
@@ -561,25 +566,75 @@ function writeIndexDTS (apis) {
   writeLine(contents, 'import \'./shim-icon-set\'')
   writeLine(contents, 'import \'./shim-lang\'')
 
-  writeFile(
-    resolvePath('index.d.ts'),
-    prettier.format(contents.join(''), { parser: 'typescript' })
-  )
+  return {
+    header: headerContents.join(''),
+    body: contents.join('')
+  }
+}
+
+const ts = require('typescript')
+
+/**
+ * @throws {Error} if TypeScript validation fails
+ */
+function ensureTypeScriptValidity () {
+  const tsConfigPath = ts.findConfigFile(distRoot, ts.sys.fileExists, 'tsconfig.json')
+  if (!tsConfigPath) {
+    throw Error(resolvePath('tsconfig.json') + ' not found')
+  }
+  const { config } = ts.readConfigFile(tsConfigPath, ts.sys.readFile)
+  config.compilerOptions.noEmit = true
+  const { options, fileNames, errors } = ts.parseJsonConfigFileContent(config, ts.sys, distRoot)
+
+  const program = ts.createProgram({ options, rootNames: fileNames, configFileParsingDiagnostics: errors })
+  const emitResult = program.emit()
+  const diagnostics = [ ...ts.getPreEmitDiagnostics(program), ...emitResult.diagnostics ]
+  if (diagnostics.length === 0) {
+    return
+  }
+
+  /** @type {ts.FormatDiagnosticsHost} */
+  const formatDiagnosticsHost = {
+    getCurrentDirectory: ts.sys.getCurrentDirectory,
+    getCanonicalFileName: file => file,
+    getNewLine: () => ts.sys.newLine
+  }
+  const error = new Error(ts.formatDiagnosticsWithColorAndContext(diagnostics, formatDiagnosticsHost))
+  error.name = 'TypeScriptError'
+  throw error
 }
 
-module.exports.generate = function (data) {
+module.exports.generate = async function (data) {
   const apis = data.plugins
     .concat(data.directives)
     .concat(data.components)
 
   try {
-    copyPredefinedTypes(typeRoot)
-    writeIndexDTS(apis)
+    await Promise.all(copyPredefinedTypes(typeRoot))
+
+    const { header, body } = getIndexDts(apis)
+    const formattedBody = prettier.format(body, { parser: 'typescript' })
+
+    // The header contains stuff that breaks TS checking.
+    // So, write only the body at first to check the validity
+    await writeFile(resolvePath('index.d.ts'), formattedBody)
```

Comment:
Yes, and I believe it's happening because there is only one `@ts-ignore`. I'll try with `@ts-ignore` for each related line to see if there will be another problem.

Reply:
Yes, `ts-ignore` only works for the following line, we need 3 of them there

Reply:
That didn't work. It goes deep into their type definitions and says stuff like "can't find `electron-builder`". We can't use `--skipLibCheck` either because that would disable our own `.d.ts` files too. That doesn't make sense, but that's what TS offers 🤷 So, the current solution is what we have, which is fine.